### PR TITLE
Feature/Add data processing models

### PR DESCRIPTION
**Pull request by blessedjasonmwanza**
## Feature/Add data processing models
In this update as per [guide](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md);
- Insert & create ```users```, ```posts``` and ```comments``` dummy data using ```rails c``` command line
- In ```posts``` add
  - A method that updates the posts counter for a user.
  - A method that returns the 5 most recent comments for a given post.
- In ```Comments```
  - A method that updates the comments counter for a post.
- Likes
  - A method that updates the likes counter for a post.
- Create ```current_user``` method  to return first user from the db

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class ApplicationController < ActionController::Base
```
Comment (Lordkaito) [inline app/controllers/application_controller.rb:1]: - [ ] Please make sure to add relations for your user model. You need to set `has_many :posts`, `has_many :comments` and `has_many :likes` to your model.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,3 +1,11 @@
 class Post < ApplicationRecord
-  belongs_to :author_id
+  belongs_to :author, class_name: 'User'
```
Comment (Lordkaito) [inline app/models/post.rb:2]: - [ ] As mentioned in the user model, please add the necessary `has_many` relationships. Here you would need to add this relation to comments and likes, so please make sure to work on this issue.

**Review by Lordkaito — CHANGES_REQUESTED** [review]
Hi @blessedjasonmwanza, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lordkaito — CHANGES_REQUESTED** [review]
Hi @blessedjasonmwanza, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by blessedjasonmwanza** [comment]
Hi @Lordkaito,

Thank you a lot for your insightful feedback. I'm on to making the requested changes right away.

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @blessedjasonmwanza , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**